### PR TITLE
WT-7815: Properly initialize prev_upd_ts for ordered timestamp assertion.

### PR DESCRIPTION
**Pull request by dgottlieb**

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -890,6 +890,14 @@ __txn_commit_timestamps_usage_check(WT_SESSION_IMPL *session, WT_TXN_OP *op, WT_
           __wt_timestamp_to_string(op_ts, ts_string[0]),
           __wt_timestamp_to_string(prev_op_durable_ts, ts_string[1])));
 
+    if (FLD_ISSET(ts_flags, WT_DHANDLE_TS_ORDERED) && prev_op_durable_ts > 0 && !txn_has_ts)
```
Comment (quchenhao) [inline src/txn/txn.c:893]: We use WT_TS_NONE for 0 timestamp.

Comment (quchenhao) [reply src/txn/txn.c:893]: I would us prev_op_durable_ts != WT_TS_NONE in this case.

**Comment by quchenhao** [comment]
The s_all test is failing in evergreen. Goto ~/wireditger/dist and run ./s_all the fix the code format.

File: src/include/serial_inline.h [hunk 2]
```diff
@@ -229,6 +229,9 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_PAGE *page
     upd = *updp;
     *updp = NULL;
     prev_upd_ts = WT_TS_NONE;
+#ifdef HAVE_DIAGNOSTIC
+    prev_upd_ts = upd->prev_durable_ts;
+#endif
```
Comment (quchenhao) [inline src/include/serial_inline.h:234]: I see what is wrong here. May be we can remove line 231 and directly initialize it to upd->prev_durable_ts. There is no need to wrap it in HAVE_DIAGNOSTIC macro.

Comment (dgottlieb) [reply src/include/serial_inline.h:234]: That's the first thing I tried. I learned that `__wt_update::prev_durable_ts` is defined in the `HAVE_DIAGNOSTIC` ifdef
https://github.com/mongodb/mongo/blob/e2395982273ecf05cb5dd9373ed106e025fb4916/src/third_party/wiredtiger/src/include/btmem.h#L1109

File: src/txn/txn.c (hunk 1 above)
Comment (dgottlieb) [reply src/txn/txn.c:893]: Done

File: src/include/serial_inline.h (hunk 2 above)
Comment (quchenhao) [reply src/include/serial_inline.h:234]: That's correct.

**Review by quchenhao — APPROVED** [review]
LGTM

**Pull request merged**